devito/symbolics/queries.py

```diff
@@ -40,7 +40,7 @@ def q_function(expr):
 
 
 def q_terminal(expr):
-    return expr.is_Symbol or expr.is_Indexed
+    return not isinstance(expr, int) and (expr.is_Symbol or expr.is_Indexed)
```

Comment:
the fact that all of a sudden we need this here, is very suspicious. What's going on?

also, you should use is_integer, but we should really try our best to avoid changing this piece of code

Reply:
We need this because we now have `int` arguments in calls

Reply:
just use ``Integer(13)`` in the Call arguments. Then it is a sympy expression and it should just work

Reply:
It wasn't the call sorry - it was the array dimension. Using `Integer` there works